packages/content-management/table_list_view_table/src/table_list_view_table.tsx

```diff
@@ -91,9 +91,10 @@ export interface TableListViewTableProps<
    */
   editItem?(item: T): void;
   /**
-   * Handler to set edit action visiblity per item.
+   * Handler to set edit action visiblity, and content editor readonly state per item.
    */
-  showEditActionForItem?(item: T): boolean;
+  itemIsEditable?(item: T): boolean;
```

Comment:
Why is itemIsEditable dashboard implemenation checking for `item.managed`? Shouldn't that information already be in `item`? So instead of this change, just check that item.managed and showEditActionForItem

Reply:
It made sense to me to unify the callback into one function that determined if the `itemIsEditable` for both the edit action + the inspect item action.
1. If the table list view gets any more edit functionality in the future, the same function can be reused.
2. The generic type `UserContentCommonSchema` isn't necessarily a saved object. We _could_ require `UserContentCommonSchema` to have a `managed?: boolean` field
3. It allows the implementor to determine for themselves what makes an item readonly or managed etc. For instance, Visualizations have a `readonly` attribute.

We could leave the function as `itemIsEditable`, and also have a default implementation that checks `managed` state.

Reply:
The problem with leaving `managed` behavior up to each implementation is that will result in inconsistent behavior across Kibana and potential bugs as teams forget to add checks for `managed`.

I would be in favor of adding `managed` to `UserContentCommonSchema`. Then the table component can provide consistent behavior for `managed` flag in a single location.

Reply:
I can add `managed` to `UserContentCommonSchema` and check it within the table list view, but it's important to note that it won't automatically make any managed saved objects non-editable. This is because the implementor is also in charge of  actually fetching the objects, and mapping them to `UserContentCommonSchema`.

Reply:
Would it make sense to make `managed` required? That why tslint would fail if its not provided?

Reply:
I was thinking about that, but it would mean we'd have to update all usages of the table + ensure that everywhere that puts saved objects into a table list view can properly get access to the managed property. I'd say that's overkill because so far the managed property is only used by integration Dashboards - plus it can only be set programmatically.

Reply:
I moved the managed checks into `tableListView` in c895a5732f7a,

Reply:
I have some concern on the indirections on how the "editabe" state is getting.

Initially, the way to indicate if a row was editable was to pass (or not) the `editItem` handler prop. Very simple and straigthforward.  Then Drew added the `showEditActionForItem` prop (not sure it was needed). Then Nathan added the `showEditActionForItem` handler...

Now I see that you add this logic

```ts
const isEditable = !item.managed && (itemIsEditable?.(item) ?? true);
```

With that now the table list view has an _internal_ business rule and the isEditable state is not full controlled by the consumer.

Then I read "_so far the managed property is only used by integration Dashboards_".

So we are adding a business rule in a "dumb" component for a single consumer. Unless there is an actual pattern across multiple content types I would not do it.

_If_ we were to add that rule, what I would expect is

```ts
// If the SO has a `managed` prop set to `true` it is never editable
const isEditable = item.managed ? false : (itemIsEditable?.(item) ?? true);
```

Reply:
@sebelga, there _will_ be a pattern for this `managed` state making line items for all consumers un-editable in the future. 

As I see it, I can either:
1. Revert the change in c895a57, which will mean the consumer is always in charge of the line item's editable state
2. Implement your suggestion which ensures that consumers cannot override the behaviour of the `managed` prop. 

I don't have a particular preference here, what would you prefer?

Reply:
I prefer option 2.

Wasn't there mention of 'by reference' saved objects in some of the managed dashboards? That would require support for managed saved objects for other types as well. Having a consistent behavior across listing pages would be ideal.

Reply:
>  there will be a pattern for this managed state

Are you saying that this is a new top level SO property handled by core available on all saved object? That's great to hear 👍 

Sorry about mentioning Drew in my comment above. I went too quick when looking at the line change in the IDE

<img width="791" alt="Screenshot 2023-09-20 at 15 36 30" src="https://github.com/elastic/kibana/assets/2854616/4a984f28-217a-4242-950a-8e1bc027d9c8">


Regarding the `showEditActionForItem`, looking at it again it seems to be a tech debt that we have. It will have to be removed in favor of using the existing `rowItemActions` (which need to support the `edit` action).

Thanks for clarifying, I agree then to go with option 2. Could you update the comments on top of the `showEditActionForItem` and `editItem` props to indicate that those will not have any effect if the content item has the `managed` prop set to `true`?

Slightly off topic:
Do all SO that have the `managed` prop set to `true` automatically get the `managed` tag (in their `references` array)?

Reply:
Okay, I've made the `managed` state override the `itemIsEditable` callback, and have added a comment to explain that in 4679bb1395d.

Yes, the `managed` state is a new top level SO property handled by core, thanks to @TinaHeiligers for that! And I'm not quite sure how the managed tag appears, but I assume it's more of a holdover from before the new property. Tina would know more.